File: php/3-largest-prime-factor.php
<?php
$naturalNumber = 600851475143;
function isAPrimeNumber(float $naturalNumber): bool
{
    if (0 === $naturalNumber % 2)
    {
        return false;
    }
    for ($divisor = 2; $naturalNumber > $divisor; ++$divisor) {
        if (0 === $naturalNumber % $divisor && $naturalNumber != $divisor) {
            return false;
        }
    }
    return true;
}

function isAFactorOfNumber(float $naturalNumber, float $divisor): bool
{
    return 0 === $naturalNumber % $divisor;
}

function getLargestPrimeFactor(float $naturalNumber)
{
    $largestPossiblePrimeNumber = floor(sqrt($naturalNumber / 2));
    for ($ctr = $largestPossiblePrimeNumber; 3 < $ctr; $ctr -= 2) {
        if (isAPrimeNumber($ctr) && isAFactorOfNumber($naturalNumber, $ctr)) {
            return $ctr;
        }
    }

    return 0;
}
var_dump(getLargestPrimeFactor(600851475143));

File: python/3-largest-prime-factor.py
import math

given_number = 600851475143


def is_a_factor_of_number(dividend: float, divisor: float):
    return False if 0 != dividend % divisor else True


def is_a_prime_number(number: float):
    if 0 == number % 2 and math.sqrt(number).is_integer():
        return False
    counter = number - 1
    while counter > 2:
        if number % counter == 0:
            return False
        counter = counter - 1
    return True


def largest_prime_factor(number, counter=2):
    while counter < math.ceil(number / 2):
        if is_a_factor_of_number(number, counter):
            return round(number / counter)
        counter = counter + 1
    return 1


print(largest_prime_factor(given_number))
